docs/core/transactions.md

```diff
@@ -0,0 +1,80 @@
+# Transactions
+
+## Prerequisites
+
+* [Anatomy of an SDK Application](./app-anatomy.md)
+
+## Synopsis
+
+This document describes how various components are defined to enable transactions. It also describes how transactions are generated.
+
+1. [Transactions](#transactions)
+2. [Transaction Definition](#transaction-definition)
+3. [CLI and REST Interfaces](#cli-and-rest-interfaces)
+4. [Messages](#messages)
+5. [Transaction Generation](#transaction-generation)
+6. [Handlers](#handlers)
+
+
+## Transactions
+
+[**Transactions**](https://github.com/cosmos/cosmos-sdk/blob/master/types/tx_msg.go#L36-L43)  are objects created by end-users to trigger state changes in the application. Specifically, they are comprised of metadata held in [contexts](./context.md) and [messages](../building-modules/messages-and-queries.md) that trigger state changes within a module through the module's Handler.
+
+When users want to interact with an application and make state changes (e.g. sending coins), they create transactions. Each of a transaction's messages must be signed using the private key associated with the appropriate account(s), and then the transaction is broadcasted to the network. A transaction must then be included in a block, validated, and approved by the network through the consensus process. To read more about the lifecycle of a transaction, click [here](../basics/tx-lifecycle.md).
+
+## Transaction Definition
+
+The transaction objects themselves are SDK types that implement the [`Tx`](https://github.com/cosmos/cosmos-sdk/blob/master/types/tx_msg.go#L34-L41) interface and include an encoder and decoder:
+
+* **GetMsgs:** unwraps the transaction and returns a list of its message(s) - one transaction may have one or multiple [messages](../building-modules/messages-and-queries.md#messages), which are defined by module developers.
```

Comment:
There is a problem with the description of these 4 methods. You are conflating the description of the interface with the description of one possible implementation.

The description of the interface just explains what the expected behaviour is (e.g. `ValidateBasic` are for stateless checks, triggered when ...).

The description of one implementation explains the specific of these checks (e.g. `ValidateBasic` checks signatures and fees). 

I'm not against including one example, but it should be clearly separated. Also if you do include an example, say in what module it is implemented and link to it.